src/Events/Calendar_Embeds/Controller.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Manages the External Calendar Embeds Feature.
+ *
+ * @since TBD
+ *
+ * @package TEC\Events\Calendar_Embeds
+ */
+
+namespace TEC\Events\Calendar_Embeds;
+
+use TEC\Common\Contracts\Provider\Controller as Controller_Contract;
+
+/**
+ * Class Controller
+ *
+ * @since TBD
+
+ * @package TEC\Events\Calendar_Embeds
+ */
+class Controller extends Controller_Contract {
+
+	/**
+	 * @inheritDoc
+	 */
+	public function do_register(): void {
+		$this->container->singleton( Calendar_Embeds::class );
+
+		$this->add_actions();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function unregister(): void {
+		$this->remove_actions();
+	}
+
+	/**
+	 * Add actions for the feature.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function add_actions() {
+		add_action( 'init', [ $this, 'register_post_type' ] );
+		add_action( 'tec_events_after_import_submenu_created', [ $this, 'register_menu_item' ] );
```

Comment:
Since this is the main controller dont hook specific actions here which in their callback they just make an instance and they are handled elsewhere. It just adds extra code without a point.

Instead: here register the Calendar_Embeds class (which i would rename to Post_type or something) which should also be a controller and there hook those and handle those.

Reply:
Same here. This is how we use controllers elsewhere, so let's discuss with @bordoni to make sure things are uniform.